snapcraft/internal/lxd.py

```diff
@@ -34,19 +36,36 @@
 _PROXY_KEYS = ['http_proxy', 'https_proxy', 'no_proxy', 'ftp_proxy']
 
 
-class Cleanbuilder:
+def _get_tar_filter(tar_filename):
+    def _tar_filter(tarinfo):
+        fn = tarinfo.name
+        if fn.startswith('./parts/') and not fn.startswith('./parts/plugins'):
+            return None
+        elif fn in ('./stage', './prime', tar_filename):
+            return None
+        elif fn.endswith('.snap'):
+            return None
+        return tarinfo
+    return _tar_filter
 
-    def __init__(self, snap_output, tar_filename, project_options,
-                 remote=None):
+
+class Containerbuild:
+
+    def __init__(self, snap_output, source_folder, project_options,
```

Comment:
That would result in "__init__() takes 1 positional argument but 7 were given" because all arguments don't have names. What's your intention here?